plugins/modules/idrac_network_attributes.py

```diff
@@ -0,0 +1,534 @@
+                                      job_status=job_resp)
+            elif job_state in ["Starting", "Scheduled", "Scheduling"]:
+                if self.module.check_mode:
+                    self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+                delete_job(self.idrac, job_id)
+                self.module.exit_json(msg=SUCCESS_CLEAR_PENDING_ATTR_MSG, changed=True)
+        if self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+        self.idrac.invoke_request(clear_pending_uri, "POST", data="{}", dump=False)
+        self.module.exit_json(msg=SUCCESS_CLEAR_PENDING_ATTR_MSG, changed=True)
+
+    def perform_operation(self):
+        oem_network_attributes = self.module.params.get('oem_network_attributes')
+        job_wait = self.module.params.get('job_wait')
+        job_wait_timeout = self.module.params.get('job_wait_timeout')
+        payload = {'Attributes': oem_network_attributes}
+        oem_links = get_dynamic_uri(self.idrac, self.network_device_function_id, 'Links')
+        oem_uri = oem_links.get('Oem').get('Dell').get('DellNetworkAttributes').get('@odata.id')
+        apply_time_setting = self.apply_time(oem_uri)
+        if apply_time_setting:
+            payload.update({"@Redfish.SettingsApplyTime": apply_time_setting})
+
+        patch_uri = get_dynamic_uri(self.idrac, oem_uri).get('@Redfish.Settings', {}).get('SettingsObject', {}).get('@odata.id')
+        response = self.idrac.invoke_request(method='PATCH', uri=patch_uri, data=payload)
+        invalid_attr = self.extract_error_msg(response)
+        job_tracking_uri = response.headers["Location"]
+        job_resp, error_msg = wait_for_idrac_job_completion(self.idrac, job_tracking_uri,
+                                                            job_wait=job_wait,
+                                                            wait_timeout=job_wait_timeout)
+
+        if error_msg:
+            self.module.exit_json(msg=error_msg, failed=True)
+        job_resp = remove_key(job_resp.json_data, regex_pattern='(.*?)@odata')
+        return job_resp, invalid_attr
+
+
+class NetworkAttributes(IDRACNetworkAttributes):
+    def __init__(self, idrac, module, base_uri):
+        super().__init__(idrac, module, base_uri)
+
+
+def perform_operation_for_main(module, obj, diff, _invalid_attr):
+    if diff:
+        if module.check_mode:
+            module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+        else:
+            job_resp, invalid_attr = obj.perform_operation()
+            if job_resp.get('JobState') == "Completed":
+                msg = SUCCESS_MSG if not invalid_attr else VALID_AND_INVALID_ATTR_MSG
+            else:
+                msg = SCHEDULE_MSG
+            module.exit_json(msg=msg, invalid_attributes=invalid_attr,
+                             job_status=job_resp, changed=True)
+    else:
+        module.exit_json(msg=NO_CHANGES_FOUND_MSG, invalid_attributes=_invalid_attr)
```

Comment:
Not erroring out if invalid attributes?

Reply:
If user has given only invalid attributes, idrac throws HTTP error, if user has given mix of valid and invalid attribute then this come into picture, then valid attributes will get applied and invalid attributes will be shown under `invalid_attributes`.

Reply:
if valid attributes are applied, how can it be NO CHANGES?

Reply:
In idempotency scenario, when user is apply same invalid + valid attributes, that time msg will be No change..., and it'll contain invalid_attr under `invalid_attributes`